### ISSUE TITLE
Float parsing failure

### ISSUE THREAD

**Issue by royzer** [googleapis/nodejs-spanner#764]
Getting the following error when I try to upsert a JS Number into a FLOAT64 column when using upsert() method -  `(new require('@google-cloud/spanner').Spanner()).instance().database().table('TableName').upsert(data)`

```
Error: 9 FAILED_PRECONDITION: Invalid value for column lat in table TableName: Expected FLOAT64.
```

Few stuffs I had tried so far,

1. Tried with parsing the float using `require('@google-cloud/spanner').Spanner.float()`, as suggested in https://github.com/googleapis/google-cloud-node/issues/2307. Failed with the error mentioned above. Following was the data passed to upsert() method.

```
{
  "lat": {
    "value": 36.124309
  },
  "lng": {
    "value": -115.124401
  },
}
```

2. Tried without parsing, with valid FLOAT values. Succeeded. Following was the data passed to upsert() method.

```
{
  "lat": 36.124309,
  "lng": -115.124401,
}
```

3.  Tried without parsing, with INT values to FLOAT64 column. Failed with the error mentioned above. Following was the data passed to upsert() method.

```
{
  "lat": 0,
  "lng": -115.124401,
}
```
Basically, I'm in a situation where I cannot write any FLOAT64 data to any of my tables. Can someone help please ?

#### Environment details

  - OS:
  - Node.js version: 8.16.2
  - npm version: 6.4.1
  - `@google-cloud/spanner` version: 4.4.1

**Comment by olavloite** [comment]
@royzer Could you add an example of how exactly your `data` variable is created / filled. If I understand  your correctly, it is an object containing two fields with the names `lat` and `lng`, both having a value of type `JS Number`. Is that correct?

**Comment by royzer** [comment]
@olavloite : Thanks for you reply. lat & lng are two fields with a value of type Spanner.Float (if I'm not wrong). `Spanner.float()` method is used to parse/convert the JS Number into float just before upsert.

The relevant part of the code would look something like this.

```
const { Spanner } = require('@google-cloud/spanner')
const spannerInstance = new Spanner();

const data = {
    lat: Spanner.float(36.124309),
    lng: Spanner.float(-115.124401),
};

await spannerInstance
    .instance()
    .database()
    .table('TableName')
    .upsert(data);
```

This leads to the error
```
Error: 9 FAILED_PRECONDITION: Invalid value for column lat in table TableName: Expected FLOAT64.
```

The JSON that I had mentioned above, is what I got when logged the `data` object in console `console.log(data)`

**Comment by royzer** [comment]
My Bad. False flag. It was another reformatting that we are using in our code between float converstion & upsert, that was breaking this.
